Remove unused imports

dannyroberts opened the pull request:
Indulging in obsession.

Opening for tests.

Last few commits remove code that's no longer used or imported anywhere, as well as https://github.com/dimagi/dimagi-utils/pull/355

dannyroberts committed:
remove unused imports corehq/apps/(a-h)

dannyroberts committed:
remove unused imports corehq/apps/(h-r)

dannyroberts committed:
remove unused imports corehq/apps/(r-z)

dannyroberts committed:
remove unused imports corehq/*

dannyroberts committed:
fix import errors

dimagimon commented on corehq/apps/style/forms/widgets.py:
```diff
@@ -1,7 +1,7 @@
 from django.forms.fields import MultiValueField, CharField
 from django.forms.utils import flatatt
-from django.forms.widgets import CheckboxInput, HiddenInput, Input, RadioSelect, RadioFieldRenderer, RadioInput, TextInput, MultiWidget
+from django.forms.widgets import CheckboxInput, Input, RadioSelect, RadioFieldRenderer, RadioInput, TextInput, MultiWidget
```
E501 line too long (122 > 115 characters)

dannyroberts committed:
remove unused BootstrapMultiField, InlineColumnField

dannyroberts committed:
remove unused get_number_of_forms_of_all_types, get_number_of_forms_by_type

dannyroberts committed:
remove unused imports in migration files

dannyroberts commented:
cc: @emord

benrudolph commented:
lol this is great

On Wednesday, March 23, 2016, Daniel Roberts <notifications@github.com>
wrote:

> cc: @emord <https://github.com/emord>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/10904#issuecomment-200560130>
>

czue commented:
haha nice. how did you do this analysis? we should probably move quickly to avoid all kinds of merge conflicts, but want to make sure you're 100% confident in your process.

dannyroberts commented:
I started by running `pylint corehq/ --disable=all --enable=unused-import` to get filenames, and then relied on pycharm's own static analyzer and removed the grey imports.

I noticed a couple places where it got it wrong (pycharm can analyze `from ... import *` incorrectly if the module uses `__all__`, thus thinking that an `os` import is overshadowed by the `*` import when it's really not), and those popped up immediately and I fixed them. There's certainly a chance that inside an untested view somewhere there are other cases of this that weren't picked up by my running runserver or running the tests.

I'm not sure how to search for that though.

The reason behind this PR is that I'm working on some simple static analysis of our code to detect functions that are declared but never used, and it's easier for me to do that if we know that if there are no unused imports.

benrudolph commented:
:bomb:

czue commented:
guess we're doing it live!

benrudolph commented:
danny's explanation convinced me :stuck_out_tongue:
